nearswap/src/twap.rs

```diff
@@ -0,0 +1,189 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::{U128, U64};
+use near_sdk::collections::{LookupMap, Vector};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, Balance, Timestamp};
+use std::convert::{TryFrom,TryInto};
+
+use std::fmt;
+use crate::constants::*;
+
+#[derive(Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize, Copy)]
```

Comment:
Why do you implement `Serialize, Deserialize` ? This is only needed if we return this structure to the user. But it's internal structure. So only Borsh* is needed.
Also, `Clone` should be only for test (look for cfg = test) attribute

Reply:
yes, removing.
ok looking

Reply:
removed clone from Twap but cannot remove from Observation because it is being cloned in Twap.

Reply:
removed clone